app/questionnaire/location.py

```diff
@@ -1,53 +1,44 @@
+from __future__ import annotations
+from dataclasses import dataclass
+
 from flask import url_for
 
 
+@dataclass
 class Location:
+    """
+        Store a location in the questionnaire.
 
-    def __init__(self, block_id):
-        self.block_id = block_id
-
-    def __eq__(self, other):
-        """
-        Check to see if two locations are equal.
-        Two answers are considered to be equal if their dictionary representations equal one another.
+        block_id: The id of the current block. This could be a block inside a list collector
+        list_item_id: The list_item_id if this location is associated with a list
+        list_name: The list name
+    """
 
-        :param other: An answer to compare
-        :return: True if both answers match, otherwise False.
-        """
-        return isinstance(other, Location) and self.__dict__ == other.__dict__
+    block_id: str
+    list_name: str = None
+    list_item_id: str = None
 
     def __hash__(self):
         return hash(frozenset(self.__dict__.values()))
 
-    def __str__(self):
-        """
-        String representation of the location, handy for debug messages
-
-        :return:
-        """
-        return '{}'.format(self.block_id)
-
-    def __repr__(self):
-        """
-        String representation of the location, handy for debug messages
-
-        :return:
-        """
-        return str(self)
-
     @classmethod
-    def from_dict(cls, location_dict):
+    def from_dict(cls, location_dict: dict):
```

Comment:
These should use `Dict` from `typing`  (and again in the return type for `from_json`). These will fail if you run it through mypy at the moment

Reply:
I think we should use mypy so @benhowes can finally be free.